File: app/src/main/java/com/krunal3kapadiya/popularmovies/data/api/MovieApiInterface.java
package com.krunal3kapadiya.popularmovies.data.api;


import com.krunal3kapadiya.popularmovies.data.model.MovieResponse;
import com.krunal3kapadiya.popularmovies.data.model.ReviewsResponse;
import com.krunal3kapadiya.popularmovies.data.model.TrailerResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by Krunal on 8/5/2017.
 */

public interface MovieApiInterface {

    @GET("movie/popular")
    Call<MovieResponse> getPopularMoviesList(@Query("api_key") String apiKey);

    @GET("movie/top_rated")
    Call<MovieResponse> getTopRatedMovies(@Query("api_key") String apiKey);

    @GET("movie/{id}/videos")
    Call<TrailerResponse> getMovieTrailers(@Path("id") long id, @Query("api_key") String apiKey);

    @GET("movie/{id}/reviews")
    Call<ReviewsResponse> getMovieReviews(@Path("id") long id, @Query("api_key") String apiKey);
}

File: app/src/main/java/com/krunal3kapadiya/popularmovies/data/api/MovieApiInterface.kt
package com.krunal3kapadiya.popularmovies.data.api

import com.krunal3kapadiya.popularmovies.data.model.CastResponse
import com.krunal3kapadiya.popularmovies.data.model.MovieResponse
import com.krunal3kapadiya.popularmovies.data.model.ReviewsResponse
import com.krunal3kapadiya.popularmovies.data.model.TrailerResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface MovieApiInterface {

    @GET("movie/popular")
    fun getPopularMoviesList(@Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("movie/top_rated")
    fun getTopRatedMovies(@Query("api_key") apiKey: String): Call<MovieResponse>


    @GET("movie/now_playing")
    fun getNowPlayingMovies(@Query("api_key") apiKey: String): Call<MovieResponse>

    @GET("movie/{id}/videos")
    fun getMovieTrailers(@Path("id") id: Long, @Query("api_key") apiKey: String): Call<TrailerResponse>

    @GET("movie/{id}/reviews")
    fun getMovieReviews(@Path("id") id: Long, @Query("api_key") apiKey: String): Call<ReviewsResponse>

    @GET("movie/{id}/casts")
    fun getCastList(@Path("id") id: Long, @Query("api_key") apiKey: String): Call<CastResponse>

    @GET("movie/upcoming")
    fun getUpComingMovies(@Query("api_key") apiKey: String): Call<MovieResponse>

    fun getMovieByYear(apI_KEY: String): Call<MovieResponse>
}
